main.py

```diff
@@ -15,39 +19,133 @@ def home():
 def selfie():
     return send_file("pictures/screenshot.jpeg", mimetype="image/jpeg")
 
-
-@app.route("/recipies", methods=["GET"])
-def recipies():
-    cippie_db = sqlite3.connect("cippedb.db")
-    cursor = cippie_db.cursor()
-    recipe = cursor.execute("SELECT food_name FROM recipies")
-    return jsonify({"status": "success", "data": recipe.fetchall()})
+    
+@app.route("/recipies", methods=["GET", "POST"])
+def get_recipies():
+    if request.method == "POST":
+        data = request.get_json()
+        try:
+            recipie = Recipie(food_name = data["name"], description = data["description"], top_image = data["top image"], bottom_image = data["bottom image"])
+            recipie.save()
+            return jsonify({"status": "success", "data": data}), 201
+        except:
+            return jsonify({"status": "error", "error": {"Important keys(in this order)": ["name", "description", "top image", "bottom image"]}}), 404
+    query = Recipie.select().dicts()
+    all_recipies = []
+    for item in query:
+        all_recipies.append(item)
+    return jsonify({"status": "success", "data": all_recipies})
 
 
 @app.route("/recipies/<name>", methods=["GET"])
 def get_recipie(name):
-    cippie_db = sqlite3.connect("cippedb.db")
-    cursor = cippie_db.cursor()
-    recipies = cursor.execute(
-        "SELECT food_name, description, top_image, bottom_image FROM recipies WHERE food_name=? ORDER BY food_name ASC LIMIT 1", 
-        (name, ))
-    return jsonify({"status": "success", "data": recipies.fetchall()})
+    query = Recipie.select().where(Recipie.food_name == name).tuples()
+    one_recipie = []
+    for item in query:
+        one_recipie.append(item)
+    return jsonify({"status": "success", "data": one_recipie})
```

Comment:
I don't think there is a need for the `"one_recipie"` array since we are returning just one item

Reply:
It'll return an object not the data 
That was the complaint i had that i told you

Reply:
I see. Thoughts @yudori?

Reply:
Yeah, if you're returning only one recipe, use get instead of select. Try something like

```python
recipe = Recipie.get(Recipie.food_name == name)
return jsonify({"status": "success", "data": recipe})
```

Reply:
@yudori will it print all the columns or isn't it just one colume i.e food_name?

Reply:
```
13
[2020-12-24 13:14:01,338] ERROR in app: Exception on /recipies/Shinkafa2 [GET]
Traceback (most recent call last):
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/app.py", line 2447, in wsgi_app
    response = self.full_dispatch_request()
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/app.py", line 1952, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/app.py", line 1821, in handle_user_exception
    reraise(exc_type, exc_value, tb)
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/_compat.py", line 39, in reraise
    raise value
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/app.py", line 1950, in full_dispatch_request
    rv = self.dispatch_request()
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/app.py", line 1936, in dispatch_request
    return self.view_functions[rule.endpoint](**req.view_args)
  File "/Users/ella/cippe/main.py", line 44, in get_recipie
    return jsonify({"status": "success", "data": recipe})
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/json/__init__.py", line 370, in jsonify
    dumps(data, indent=indent, separators=separators) + "\n",
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/json/__init__.py", line 211, in dumps
    rv = _json.dumps(obj, **kwargs)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/__init__.py", line 234, in dumps
    return cls(
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 199, in encode
    chunks = self.iterencode(o, _one_shot=True)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 257, in iterencode
    return _iterencode(o, 0)
  File "/Users/ella/cippe/venv/lib/python3.8/site-packages/flask/json/__init__.py", line 100, in default
    return _json.JSONEncoder.default(self, o)
  File "/usr/local/opt/python@3.8/Frameworks/Python.framework/Versions/3.8/lib/python3.8/json/encoder.py", line 179, in default
    raise TypeError(f'Object of type {o.__class__.__name__} '
TypeError: Object of type Recipie is not JSON serializable
127.0.0.1 - - [24/Dec/2020 13:14:01] "GET /recipies/Shinkafa2 HTTP/1.1" 500 -
```
It will print only the id, if i use get and return this error @yudori @jirevwe

Reply:
```
@app.route("/recipies/<name>", methods=["GET"])
def get_recipie(name):
    recipe = Recipie.get(Recipie.food_name == name)
    print(recipe)
    return jsonify({"status": "success", "data": recipe})
```
This is what I wrote, so i need to iterate through to get the values

Reply:
No, no need to iterate, you just need to convert it to json.

Reply:
This should help http://docs.peewee-orm.com/en/latest/peewee/playhouse.html#model_to_dict

Reply:
I really don't know what the problem is but I checked and I'm getting a name error @yudori
```
 File "/Users/ella/cippe/main.py", line 53, in get_recipie
    return jsonify({"status": "success", "data": model_to_dict(query, backrefs=True)})
NameError: name 'model_to_dict' is not defined
```

Reply:
You need to import the function like so:
```python
from playhouse.shortcuts import model_to_dict
```

Reply:
God bless you @yudori

Reply:
@yudori. changing type to `ingredient_type` might pose a problem because the db column uses type. Do i need to change that

Reply:
You'll need to re-create the table so that the database re-writes the column with the new name.

Reply:
Alright @yudori